controller/dataplane/owned_custom_plugins.go

```diff
@@ -0,0 +1,81 @@
+package dataplane
+
+import (
+	"fmt"
+	"strings"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/kong/gateway-operator/internal/utils/dataplane"
+	"github.com/kong/gateway-operator/pkg/consts"
+	k8sresources "github.com/kong/gateway-operator/pkg/utils/kubernetes/resources"
+)
+
+type customPlugin struct {
+	// Name of the KongPluginInstallation resource.
+	Name string
+	// ConfigMapNN is the namespace/name of the ConfigMap that contains the plugin.
+	ConfigMapNN types.NamespacedName
+	// Generation is the generation of the KongPluginInstallation that contains the plugin.
+	Generation int64
+}
+
+func withCustomPlugins(customPlugins []customPlugin) k8sresources.DeploymentOpt {
```

Comment:
WDYT about changing the signature here so that it accepts a variadic list instead of a slice? This way we don't need to provide `nil` when no custom Plugins are to be configured.